src/services/resource.js

```diff
@@ -354,13 +354,24 @@ const filterResourcesForSync = (
         };
       }
 
+      const resourceSourceFields = (() => {
+        // I am seeing this occaisionally come through as a string, like a SQL array literal
```

Comment:
This is going to end up with bad data in the db, I think the path with the error needs to be found. Just short circuiting it is not a good idea.